### PR TITLE
Chloromorph Permits Forced Sleep

### PR DESCRIPTION
**Pull request by OmniscientQ**
#### Summary
Balance "Chloromorph can be activated to sleep all day"

#### Purpose of change

The Chloromorph mutation is frustrating to use, since it requires the mutant to sleep during the day in order to get any benefits from it. This encourages a Plant mutant to be nocturnal, so they can make themselves tired enough to fall asleep once the sun rises. Any effects or traits that tinker with your sleep cycle (like Less Sleep, granted by picking the Video Games hobby) makes it difficult to maintain that cycle. The mutation doesn't even make it possible to sleep during the day without a blindfold.

#### Describe the solution

This PR makes Chloromorph an activated mutation. The animal bits of the mutant are shut down as it enters Full Plant Mode. When active, it makes the mutant immune to light levels interrupting sleep, sets their comfort level high enough that they can sleep any time (still subject to restrictions about sleeping on diggable soil), and uses the same checks as anesthesia to ensure the mutant won't wake up while the sun is shining on them. If the mutant is sleeping underground or at night, the plant-coma won't be activated.

#### Describe alternatives you've considered

I thought about having the fatigue level slightly lower, so that there was some chance of waking up early or late. Set the levels such that, on average, the hibernation will last 16 or 24 hours, but a Plant mutant going into hibernation doesn't know for sure when they'll wake up.

Also considered having the mutant always sleep for 24 hours, but testing just resulted in the same schedule slippage that prompted these changes to begin with.

I also thought about making them resistant to shivering, since the mutant has to be outdoors to get the benefits of this trait. However, the animal bits are still there, even if they aren't doing much. (It's not too clear how thoroughly blended the plant and animal bits are). Any resistance to chill night air they might have been eligible for is probably offset by their lower heat output while sleeping.

Finally, I wondered whether the plant hibernation should be voluntary at all, if every sleep should be subject to these hibernation rules.

#### Testing

Tested sleeping with a basic, non-mutated human in the evac shelter. I couldn't find any differences in behavior there.

Then mutated the survivor with Chloromorph, and had them sleep outside, with the mutation turned off. It produced the exact same behavior as before my changes. The mutant cannot sleep on a bed anymore, but is very comfy in the dirt. The mutant woke up after a few hours due to sunlight levels.

Activating Chloromorph puts the mutant to sleep after about an hour, and sunlight levels don't matter. The mutant slept until nightfall, then woke up due to shivering. Spawned in a sleeping bag, put it down in that tile, then tried to sleep again. It took two hours for sleep to resume. This time, the mutant slept for a solid 24 hours. Vitamin and calorie generation were within the expected ranges.

Alarm clocks woke up the mutant without a problem, if the option were selected. With the Poor Hearing and Heavy Sleeper traits (both also in the Plant line), the mutant consistently slept through the alarm for three or four hours.

The mutant successfully woke up when attacked.

EDIT: After changing the soporific effects of this change to be dependent on sunlight, the mutant now wakes up shortly after nightfall.

#### Additional context

Even if the overall idea is rejected, I still think that the Thirst portion of the old Chloromorph shouldn't be tied to it being sunny. Also, the Roots mutations don't work while sleeping, so the vitamin mods should probably be added in anyway.

### CODE REVIEW

**Comment by OmniscientQ** [comment]
I didn't realize this had effectively bundled my previous PR with the JSON edits. Sorry.

File: data/json/mutations/mutations.json [hunk 1]
```diff
@@ -6164,7 +6164,7 @@
     "points": 10,
     "visibility": 7,
     "ugliness": 1,
-    "description": "Every inch of your skin is packed with chlorophyll and you have strong roots.  Sleeping on diggable soil will satisfy any hunger or thirst you might have.",
+    "description": "Every inch of your skin is packed with chlorophyll and you have strong roots.  Sleeping on diggable soil will satisfy any hunger or thirst you might have. Activate this mutation to force yourself to sleep for 24 hours.",
```
Comment (Maleclypse) [inline data/json/mutations/mutations.json:6167]:
```suggestion
    "description": "Every inch of your skin is packed with chlorophyll and you have strong roots.  Sleeping on diggable soil will satisfy any hunger or thirst you might have.  Activate this mutation to force yourself to sleep for 24 hours.",
```

Comment (Maleclypse) [inline data/json/mutations/mutations.json:6167]:
```suggestion
    "description": "Every inch of your skin is packed with chlorophyll and you have strong roots.  Sleeping on diggable soil will satisfy any hunger or thirst you might have.  Activate this mutation to force yourself to sleep for 24 hours.",
```

Comment (Maleclypse) [inline data/json/mutations/mutations.json:6167]:
```suggestion
    "description": "Every inch of your skin is packed with chlorophyll and you have strong roots.  Sleeping on diggable soil will satisfy any hunger or thirst you might have.  Activate this mutation to force yourself to sleep for 24 hours.",
```

File: src/player_hardcoded_effects.cpp [hunk 2]
```diff
@@ -125,6 +125,8 @@ static const trait_id trait_WATERSLEEP( "WATERSLEEP" );
 
 static const vitamin_id vitamin_blood( "blood" );
 static const vitamin_id vitamin_redcells( "redcells" );
+static const vitamin_id vitamin_calcium( "calcium" );
+static const vitamin_id vitamin_iron( "iron" );
```
Comment (ZhilkinSerg) [inline src/player_hardcoded_effects.cpp:129]:
```suggestion
static const vitamin_id vitamin_blood( "blood" );
static const vitamin_id vitamin_calcium( "calcium" );
static const vitamin_id vitamin_iron( "iron" );
static const vitamin_id vitamin_redcells( "redcells" );
```

**Review by Zireael07 — COMMENTED** [review]
Json changes look like a bad merge?

**Comment by OmniscientQ** [comment]
> 

I am not surprised... I don't work with git very often. I tried to pull the latest changes from the main repo to my drive, and naturally screwed something up. What's the best way to fix it?

**Review by Zireael07 — COMMENTED** [review]
Bad merge?

File: data/json/mutations/mutations.json [hunk 3]
```diff
@@ -6164,7 +6164,7 @@
     "points": 10,
     "visibility": 7,
     "ugliness": 1,
-    "description": "Every inch of your skin is packed with chlorophyll and you have strong roots.  Sleeping on diggable soil will satisfy any hunger or thirst you might have.",
+    "description": "Every inch of your skin is packed with chlorophyll and you have strong roots.  Sleeping on diggable soil will satisfy any hunger or thirst you might have.  Activate this mutation to force yourself to sleep for 24 hours.",
```
Comment (kevingranade) [inline data/json/mutations/mutations.json:6167]:
```suggestion
    "description": "Every inch of your skin is packed with chlorophyll and you have strong roots.  Sleeping on diggable soil will satisfy any hunger or thirst you might have.  Activate this mutation to force yourself to sleep until nightfall,
```

File: src/player_hardcoded_effects.cpp [hunk 4]
```diff
@@ -937,7 +939,7 @@ static void eff_fun_sleep( Character &u, effect &it )
         it.mod_intensity( 1 );
     }
 
-    if( u.has_effect( effect_narcosis ) && u.get_fatigue() <= 25 ) {
+    if( ( u.has_effect( effect_narcosis ) ) && u.get_fatigue() <= 25 ) {
```
Comment (kevingranade) [inline src/player_hardcoded_effects.cpp:942]:
```suggestion
    if( u.has_effect( effect_narcosis ) && u.get_fatigue() <= 25 ) {
```

File: data/json/mutations/mutations.json [hunk 5]
```diff
@@ -6164,7 +6164,7 @@
     "points": 10,
     "visibility": 7,
     "ugliness": 1,
-    "description": "Every inch of your skin is packed with chlorophyll and you have strong roots.  Sleeping on diggable soil will satisfy any hunger or thirst you might have.",
+    "description": "Every inch of your skin is packed with chlorophyll and you have strong roots.  Sleeping on diggable soil will satisfy any hunger or thirst you might have.  Activate this mutation to force yourself to sleep until nightfall,
```
Comment (anothersimulacrum) [inline data/json/mutations/mutations.json:6167]:
```suggestion
    "description": "Every inch of your skin is packed with chlorophyll and you have strong roots.  Sleeping on diggable soil will satisfy any hunger or thirst you might have.  Activate this mutation to force yourself to sleep until nightfall",
```

**Pull request merged**